modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/pagemem/SpeedBasedCleanPagesProtectionThrottle.java

```diff
@@ -0,0 +1,500 @@
+/*
+ * Copyright 2021 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.persistence.pagemem;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.ignite.internal.processors.cache.persistence.checkpoint.CheckpointProgress;
+import org.apache.ignite.internal.util.GridConcurrentHashSet;
+import org.apache.ignite.lang.IgniteOutClosure;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Speed-based throttle used to protect clean pages from exhaustion.
+ */
+class SpeedBasedCleanPagesProtectionThrottle {
+    /**
+     * Maximum dirty pages in region.
```

Comment:
Maximum percentage of dirty pages?

Reply:
Fixed it as 'Maximum fraction of dirty pages' as percentage should probably be in percents.